pkg/api/v1/schema.go

```diff
@@ -56,6 +56,9 @@ type DomainSpec struct {
 	// CPU allow specified the detailed CPU topology inside the vm.
 	// +optional
 	CPU *CPU `json:"cpu,omitempty"`
+	// Hugepages allow to specify desired hugepages size for the vm usage.
+	// +optional
+	Hugepages *Hugepages `json:"hugepages,omitempty"`
```

Comment:
Can't we simply express that the same way like in k8s unter `resources`?

```
apiVersion: v1
kind: Pod
metadata:
  generateName: hugepages-volume-
spec:
  containers:
  - image: fedora:latest
    command:
    - sleep
    - inf
    name: example
    volumeMounts:
    - mountPath: /hugepages
      name: hugepage
    resources:
      limits:
        hugepages-2Mi: 100Mi
  volumes:
  - name: hugepage
    emptyDir:
      medium: HugePages
```

We can still validatate that only huge-pages or memory is used.

Reply:
In the K8s context a user requests that resource, and he will explicitly get it (as a device).

In the VM context however, the user requests a resource, that is memory, and hugepages are merely a _different backend_ for the requested memory.

IOW: The broad ppicture is that the reqource requests outline what the user will get in the context of his workload. In pods he can have memory and hugepages. But for VMs he will always just get a virtual memory - which can then either be backed by hugepages or not -at least inside the VM he will only have memory.

This is why it is alright to just have a memory requests field, and not have the user to request something else than memory.

Hwoever, if we see that this can not stand over th elong run, then let's revisit the API.

Reply:
> IOW: The broad ppicture is that the reqource requests outline what the user will get in the context of his workload. In pods he can have memory and hugepages. But for VMs he will always just get a virtual memory - which can then either be backed by hugepages or not -at least inside the VM he will only have memory.

Only allowing specifying one of those should be good enough. Simple validation check. Especially from the point of view you bring up, it is strange that you specify hugepages in the domain section, since as you say inside the VM it is only memory, but you need to specify it as a guest detail.

> In the K8s context a user requests that resource, and he will explicitly get it (as a device).

Also in the vm case he explicitly asks for it. If it will not be available the vm will not be scheduled. Very explicit.

> This is why it is alright to just have a memory requests field, and not have the user to request something else than memory.

This proposed implementation in fact moves parts of the scheduling decision regarding to resources into the guest details and hides what will be used by the scheduler.

> Hwoever, if we see that this can not stand over th elong run, then let's revisit the API.
> 
>  Write

Yes.

Reply:
To be honest resource usage for hugepages will include some different validations that I wanted to avoid.
In the case when you specified hugepages:
- no memory request
- no memory limit
- hugepage request without limit
- different hugepage request and limit
I think k8s resource API does not suit well for resources without overcommit, because it leaves some corner cases, that user will reveal only after the start of the pod